amazonfreertossdk/src/main/java/software/amazon/freertos/amazonfreertossdk/AmazonFreeRTOSDevice.java

```diff
@@ -437,7 +447,7 @@ public void onServicesDiscovered(BluetoothGatt gatt, int status) {
                         Log.i(TAG, "Discovered Ble gatt services successfully. Bonding state: "
                                 +mBluetoothDevice.getBondState());
                         describeGattServices(mBluetoothGatt.getServices());
-                        if (mBluetoothDevice.getBondState() == BluetoothDevice.BOND_BONDED) {
+                        if (mBluetoothDevice.getBondState() == BluetoothDevice.BOND_BONDING) {
```

Comment:
Why is this within a bonding state ? From `onConnectionStateChange()` code it looks like `discoverServices()` will only be called when `bondState != BluetoothDevice.BOND_BONDING`

Reply:
Right, if we have successfully discovered the service, the bond_state should be correct so we can call initialize() without further check.